From f8104a112eb967cd4c2aaf57b5a59aafe7c4738f Mon Sep 17 00:00:00 2001
From: Peter Wu <peter@lekensteyn.nl>
Date: Wed, 21 Jan 2015 15:56:23 +0100
Subject: [PATCH] rules: support Logitech Unifying in Linux 3.19

Linux 3.19 changed the way in which devices get registered by
introducing a hid-logitech-hidpp module. The driver name therefore needs
to be adjusted.

Signed-off-by: Peter Wu <peter@lekensteyn.nl>
---
 rules/95-upower-csr.rules | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/rules/95-upower-csr.rules b/rules/95-upower-csr.rules
index b476660..39539ca 100644
--- a/rules/95-upower-csr.rules
+++ b/rules/95-upower-csr.rules
@@ -25,5 +25,8 @@ SUBSYSTEM!="hid", GOTO="up_unifying_end"
 ATTRS{idVendor}=="046d", ENV{UPOWER_VENDOR}="Logitech, Inc."
 ATTRS{idVendor}=="046d", ATTRS{idProduct}=="c52b", DRIVER=="logitech-djdevice", ENV{UPOWER_BATTERY_TYPE}="unifying"
 ATTRS{idVendor}=="046d", ATTRS{idProduct}=="c532", DRIVER=="logitech-djdevice", ENV{UPOWER_BATTERY_TYPE}="unifying"
+# These devices bind to the hid-logitech-hidpp module since Linux 3.19
+ATTRS{idVendor}=="046d", ATTRS{idProduct}=="c52b", DRIVER=="logitech-hidpp-device", ENV{UPOWER_BATTERY_TYPE}="unifying"
+ATTRS{idVendor}=="046d", ATTRS{idProduct}=="c532", DRIVER=="logitech-hidpp-device", ENV{UPOWER_BATTERY_TYPE}="unifying"
 ATTRS{idVendor}=="046d", ATTRS{idProduct}=="c52f", ENV{UPOWER_BATTERY_TYPE}="lg-wireless"
 LABEL="up_unifying_end"
-- 
2.6.3

